### PR TITLE
Add log for slow database lock and fix some logging

### PR DESCRIPTION
**Pull request by decster**
## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #

## Problem Summary(Required) ：

This PR adds warning log with some debug information when db's read/write lock takes a long time to lock, this helps identify lock contention issues. Also fix a logging error in TabletScheduler.

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:disappointed: **fail** : 9 / 20 (45.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/clone/TabletScheduler.java|0|6|00.00%|
|:large_blue_circle:|com/starrocks/catalog/Database.java|7|11|63.64%|
|:large_blue_circle:|com/starrocks/catalog/Replica.java|2|3|66.67%|

File: fe/fe-core/src/main/java/com/starrocks/catalog/Database.java [hunk 1]
```diff
@@ -121,7 +125,14 @@ public Database(long id, String name) {
     }
 
     public void readLock() {
+        long startMs = System.nanoTime() / 1000000;
```
Comment (Astralidea) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Database.java:128]: use tryLock same like Catalog.tryLock may be better?

Comment (decster) [reply fe/fe-core/src/main/java/com/starrocks/catalog/Database.java:128]: This PR only add some log, it did not change existing code logic. There are too many places call readLock & writeLock, we cannot change them all

**Review by Astralidea — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/catalog/Database.java [hunk 2]
```diff
@@ -145,7 +156,14 @@ public void readUnlock() {
     }
 
     public void writeLock() {
+        long startMs = System.nanoTime() / 1000000;
         this.rwLock.writeLock().lock();
+        long endMs = System.nanoTime() / 1000000;
+        if (endMs - startMs > SLOW_LOCK_MS && endMs > lastSlowLockLogTime + SLOW_LOCK_LOG_EVERY_MS) {
+            lastSlowLockLogTime = endMs;
+            LOG.warn("slow write lock db:" + id + " " + fullQualifiedName + " " + (endMs - startMs) + "ms",
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Database.java:164]: Printing the warning log when holding the lock is not a good idea.

**Review by chaoyli — APPROVED** [review]

**Pull request merged**